analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -57,23 +57,30 @@ class Analyzer(private val config: AnalyzerConfiguration) {
             RepositoryConfiguration(null)
         }
 
-        // Map of files managed by the respective package manager.
-        var managedFiles = if (packageManagers.size == 1 && absoluteProjectPath.isFile) {
+        // Map files by the package manager factory that manages them.
+        val factoryFiles = if (packageManagers.size == 1 && absoluteProjectPath.isFile) {
             // If only one package manager is activated, treat the given path as definition file for that package
             // manager despite its name.
             mutableMapOf(packageManagers.first() to listOf(absoluteProjectPath))
         } else {
             PackageManager.findManagedFiles(absoluteProjectPath, packageManagers).toMutableMap()
         }
 
-        val hasDefinitionFileInRootDirectory = managedFiles.values.flatten().any {
+        val hasDefinitionFileInRootDirectory = factoryFiles.values.flatten().any {
             it.parentFile.absoluteFile == absoluteProjectPath
         }
 
-        if (managedFiles.isEmpty() || !hasDefinitionFileInRootDirectory) {
-            managedFiles[Unmanaged.Factory()] = listOf(absoluteProjectPath)
+        if (factoryFiles.isEmpty() || !hasDefinitionFileInRootDirectory) {
+            factoryFiles[Unmanaged.Factory()] = listOf(absoluteProjectPath)
         }
 
+        // Instantiate the managers via their factories.
+        val managedFiles = factoryFiles.mapNotNull { (factory, files) ->
```

Comment:
One small detail: before it was very obvious that package manager instance are just used once and now I think it becomes a bit less obvious - I'd have slight fear this increases the chance of someone introducing instance re-use in future. 

What do you think about not keeping the manager instance, but just having one step which adjusts the 
mapping: factory -> List<DefinitionFile>

And for the mgr.resolveDependencies call, you would create a new mgr instance.
it could be extracted to a function (the name is no good i know):

fun remapDefinitionFiles(map: Map<PackageManagerFactory, File>) : Map<PackageManagerFactory, File>)